### PR TITLE
fix: HomeHero NavLink reactivity lost (#195)

### PR DESCRIPTION
**Pull request by ShenQingchuan**
Please see https://github.com/vuejs/vitepress/issues/195 for more details... (May Chinese required)

### CODE REVIEW

**Comment by ShenQingchuan** [comment]
# Something Weird

For the code below:
```typescript
const props = defineProps<{
  item: DefaultTheme.NavItemWithLink,
}>();
const propsRefs = toRefs(props);
const { props: linkProps, isExternal } = useNavLink(propsRefs.item);
```

It seems very tedious, right?
but when I changed it to this:
```typescript
const propsRefs = toRefs(defineProps<{
  item: DefaultTheme.NavItemWithLink,
}>());
const { props: linkProps, isExternal } = useNavLink(propsRefs.item);
```

the JS Console print this:

image

May this be a new bug or something wrong with my code ?

File: src/client/theme-default/components/NavLink.vue [hunk 1]
```diff
@@ -7,16 +7,16 @@
...
-const { item } = defineProps<{
-  item: DefaultTheme.NavItemWithLink
-}>()
-
-const { props: linkProps, isExternal } = useNavLink(item)
+const props = defineProps<{
+  item: DefaultTheme.NavItemWithLink,
+}>();
+const propsRefs = toRefs(props);
+const { props: linkProps, isExternal } = useNavLink(propsRefs.item);
```
Comment (kiaking) [inline src/client/theme-default/components/NavLink.vue:19]: I think we only need to stop destructuring. So following should work. Could you test with one? It's working on my env.

```suggestion
const props = defineProps<{
  item: DefaultTheme.NavItemWithLink,
}>();

const { props: linkProps, isExternal } = useNavLink(props.item);
```

**Review by kiaking — CHANGES_REQUESTED** [review]
Thank you so much for the PR! I've left a comment so please have a look. I think we only need to change `NavLink.vue` with this change.

**Comment by ShenQingchuan** [comment]
Yes. Without `toRefs` it still works, that's my mistake...

**Review by kiaking — CHANGES_REQUESTED** [review]
I don't think we need `NavDropdownLinkItem.vue ` and `navLink.ts` changes anymore. Can we delete them? 👀

**Comment by ShenQingchuan** [comment]
> I don't think we need `NavDropdownLinkItem.vue ` and `navLink.ts` changes anymore. Can we delete them? 👀

Sure, I'll refactor my commits and repush later...

**Comment by ShenQingchuan** [comment]
**OOPS !! @_@** @kiaking 

 `navLinks.ts` need to be changed...

**Why I need the argument `item` of `useNavLink(item)`  to be a `Ref<DefaultTheme.NavItemWithLink>`**:

Because the value you passed into in `HomeHero.vue` as `item` v-bind: 

```vue
<NavLink
  v-if="hasAction"
  :item="{ link: data.actionLink, text: data.actionText }"
  class="action"
/>
```
is **Not** a `ref` at all, when the locales changed, the data(from frontMatter info) did change, but the object itself didn't change at all, it performs as a constant object.

cut a long story short, it will show a bug when you follow these steps:
1. open a English version doc first
2. click another language locale link
3. and you can see the homehero button's text has changed, that's fine and what we expected.
4. but if you click the button, then we will still go to the English section, not the language we wanted.

**Comment by ShenQingchuan** [comment]
the `item` passed it into `useNavLink` need to be a `ref`, or it will never changed... 

you can set a `watch` and write a function to `console.log(item.link)` for debuging, 

and you'll see the `item.link` string value has never changed to anthoer language (such as 'zh') `/zh/xxx` but still `/xxx`

**Comment by ShenQingchuan** [comment]
I did this PR a long time ago, I just didn't remember some details, feel really embarrassed...

**Comment by kiaking** [comment]
Ah good point! Yap. To note, it's not because of how we pass props, props are reactive because `item.text` is working just fine. It's about how we're passing props to the composable function. Maybe we should have made it call it from the parent component. I might refactor this later.

But for now, thank you so much for the fix!

**Review by kiaking — APPROVED** [review]

**Pull request merged**